codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -783,7 +805,7 @@ void fireChildRead(Http2Frame frame) {
 
         void fireChildReadComplete() {
             assert eventLoop().inEventLoop();
-            assert readInProgress;
+            assert readStatus == ReadStatus.IN_PROGRESS;
```

Comment:
Is this still accurate? If the user had requested a read while in the `IN_PROGRESS` state I think we transition to the `REQUESTED` state. In fact, the `unsafe.notifyReadComplete` call right triggered next will check if we're in the `REQUESTED` state and transition back to `IN_PROGRESS`.

Reply:
@bryce-anderson I think you are right... it should be either IN_PROGRESS or REQUESTED. Let me fix this.

Reply:
@bryce-anderson PTAL https://github.com/netty/netty/pull/8639 and thanks